main.js

```diff
@@ -0,0 +1,113 @@
+//TESTS
+//testing transpose function
+if (compareArrays(transpose(transpose([[1]])),[[1]])) {
+  if(compareArrays(transpose([[1,2],[3,4]]),[[1,3],[2,4]])){
+  } else {
```

Comment:
Notice that you're nesting the comparisons: will this test throw an exception as intended if `compareArrays(transpose(transpose([[1]])),[[1]]))` returns `false`?